streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -1121,13 +1122,31 @@ public String name() {
 
             @Override
             public void apply(final StreamTask task) {
-                task.suspend();
+                try {
+                    task.suspend();
+                } catch (final Exception e) {
+                    try {
+                        task.close(false);
+                    } catch (final Exception f) {
+                        log.error("{} Closing task {} failed: ", logPrefix, task.id, f);
+                    }
+                    throw e;
```

Comment:
Btw: the original fix tackled the "close" scenario only. I am not sure, if the suspend scenario is handled properly in `0.10.2` branch -- as we don't know if there will be `0.10.2.2` release, I am not sure if it worth the time to do a backport of this patch.